zilliqa/src/p2p_node.rs

```diff
@@ -146,9 +161,14 @@ impl P2pNode {
                     .map_err(|e| anyhow!(e))?,
                     autonat_client: autonat::v2::client::Behaviour::default(),
                     autonat_server: autonat::v2::server::Behaviour::default(),
-                    kademlia: kad::Behaviour::new(peer_id, MemoryStore::new(peer_id)),
+                    kademlia_legacy: kad::Behaviour::new(peer_id, MemoryStore::new(peer_id)),
+                    kademlia: kad::Behaviour::with_config(
+                        peer_id,
+                        MemoryStore::new(peer_id),
+                        kad::Config::new(kad_protocol.clone()),
+                    ),
                     identify: identify::Behaviour::new(
-                        identify::Config::new("zilliqa/1.0.0".into(), key_pair.public())
+                        identify::Config::new(protocol_version.clone(), key_pair.public())
```

Comment:
Protocol versions do not need to match for nodes to communicate - https://github.com/libp2p/specs/blob/master/identify/README.md#protocolversion